msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/NativeAuthPublicClientApplicationAnotherAbstractTest.kt

```diff
@@ -0,0 +1,156 @@
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.e2e.tests.network.nativeauth
+
+import android.app.Activity
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.microsoft.identity.client.ILoggerCallback
+import com.microsoft.identity.client.Logger
+import com.microsoft.identity.client.PublicClientApplication
+import com.microsoft.identity.client.e2e.shadows.ShadowAndroidSdkStorageEncryptionManager
+import com.microsoft.identity.client.e2e.tests.IPublicClientApplicationTest
+import com.microsoft.identity.client.exception.MsalException
+import com.microsoft.identity.common.internal.controllers.CommandDispatcherHelper
+import com.microsoft.identity.internal.testutils.TestUtils
+import com.microsoft.identity.internal.testutils.labutils.KeyVaultFetchHelper
+import com.microsoft.identity.internal.testutils.labutils.LabConstants
+import com.microsoft.identity.internal.testutils.labutils.LabUserHelper
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery
+import com.microsoft.identity.internal.testutils.nativeauth.ConfigType
+import com.microsoft.identity.internal.testutils.nativeauth.api.models.NativeAuthTestConfig
+import com.microsoft.identity.nativeauth.INativeAuthPublicClientApplication
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+// TODO: move to "PAUSED". A work in RoboTestUtils will be needed though.
```

Comment:
Is this TODO needed?

Reply:
Removing this TODO is not inside the work of this scope thus I would be prudential here of removing it.

Reply:
Now that we are back to having one single `NativeAuthPublicClientApplicationAbstractTest` class, the TODO is still there. Do we need it? (I'm just asking because I'm not aware of the context)

Reply:
The TODO comment here is to give an idea of using new functionality of Robolectric @LooperMode(PAUSED) instead of @LooperMode(LooperMode.Mode.LEGACY).

Reply:
We can do this boost, but I'm not sure if it should be included in this pull request. If I want to solve this TODO, I'll need some extra time and background. And perhaps I should seek guidance from Sammy about this TODO.

Reply:
No worries, it was just for my understanding, I'm resolving this comment and will let @nilo-ms handle that decision with you.